src/Angor/Client/Pages/Spend.razor

```diff
@@ -85,7 +85,7 @@
                     @if (stageisActive)
                     {
 
-                        <button class="btn @((noCoinsToClaim) ? "btn-light" : "btn-success")" disabled="@noCoinsToClaim" @onclick="() => ClaimCoinsCheckPassword(stage.StageIndex)">
+                        <button class="btn @((noCoinsToClaim) ? "btn-light" : "btn-success")" disabled="@((noCoinsToClaim || !IsCheckboxSelectedForStage(stage.StageIndex)))" @onclick="() => ClaimCoinsCheckPassword(stage.StageIndex)">
```

Comment:
This can be much simplified, when we expand another stage we clear the dictionary so there will only be items for a particular stage right? so you just need to check if the dictionary `selectedUtxos` has items in it, correct me if I am wrong.

Reply:
`noCoinsToClaim` is determined by `investedCount == 0` or `stage.StagePinner == true`. 
 there might be a scenario where `selectedUtxos` has values for stage 2. In this case, the `claimCoins` option for stage 1 won't be disabled. 
 providing the `stageIndex` and the relevant method, we can handle the logic for each stage appropriately.